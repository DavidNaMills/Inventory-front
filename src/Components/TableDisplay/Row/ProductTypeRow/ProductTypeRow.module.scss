.productTypeRow{
        // width: 75%;
        display: flex;
        align-items: baseline;
        // margin-left: 20px;
        padding: 10px;
        margin-bottom: 10px;

    &__title{
        margin-right: 50px;
    }

    &__cellCont{
        display:flex;
        padding-top: 20px;

        p{
            margin-right: 50px;
        }

        p:last-of-type {
            margin-right: none;
        }
    }
    
    &__cellContPortrait{
        display:flex;
        flex-direction: column;
        padding-top: 20px;

        p{
            margin-right: 50px;
        }

        p:last-of-type {
            margin-right: none;
        }
    }
    span{
        font-weight: bold;
        margin-right: 10px;
    }
}