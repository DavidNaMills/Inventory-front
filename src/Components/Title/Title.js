import React from 'react';
import classes from './title.module.scss';


const Title = (props) => {
    const { size = null, message = '', center = false } = props;
    let element = null;

    switch (size) {
        case 'h1':
            element = <h1 className={[classes.title_h1, center ? classes.title__center : null].join(' ')} >{message}</h1>
            break;
        case 'h2':
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
        case 'h3':
            element = <h3 className={[classes.title_h3, center ? classes.title__center : null].join(' ')}>{message}</h3>
            break;
        case 'h4':
            element = <h4 className={[classes.title_h4, center ? classes.title__center : null].join(' ')}>{message}</h4>
            break;
        case 'h5':
            element = <h5 className={[classes.title_h5, center ? classes.title__center : null].join(' ')}>{message}</h5>
            break;
        case 'h6':
            element = <h6 className={[classes.title_h6, center ? classes.title__center : null].join(' ')}>{message}</h6>
            break;
        default:
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
    };

    return (
        <div>
            {element}
        </div>
    )
}

export default React.memo(Title);