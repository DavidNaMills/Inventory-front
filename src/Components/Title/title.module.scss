@import '../../styles/variables/colors.module.scss';
@import '../../styles/fonts/fontSizes.module.scss';
@import '../../styles/mixins/title.module.scss';

.title {
    @include setTitle($h2, $title-color);
    
    &_h1{
        @include setTitle($h1, $title-color);
    };

    &_h2{
        @include setTitle($h2, $title-color);
    };
    
    &_h3{
        @include setTitle($h3, $title-color);
    }

    &_h4{
        @include setTitle($h4, $title-color);
    }

    &_h5{
        @include setTitle($h5, $title-color);
    }

    &_h6{
        @include setTitle($h6, $title-color);
    }

    &__center{
        text-align: center;
    }


}