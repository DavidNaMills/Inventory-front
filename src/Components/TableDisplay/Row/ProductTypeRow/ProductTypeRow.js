import React from 'react';
import classes from './ProductTypeRow.module.scss';
import Title from '../../../Title/Title';
import {isMobileOnly} from 'react-device-detect';
import { useLocale } from '../../../../Context/LocaleContext';

const ProductTypeRow = (props) => {
    const { config, data } = props;
    const locale = useLocale();

    return (
        <div className={classes.productTypeRow}>
            <div className={classes.productTypeRow__title}>
                <Title message={data[config.main[0]]} size='h4' />
            </div>
            {!isMobileOnly &&
                <div className={classes.productTypeRow__cellCont}>
                    <p><span>{locale[config.main[1]]}:</span>{data[config.main[1]]}</p>
                </div>
            }
        </div>
    )
}

export default ProductTypeRow;