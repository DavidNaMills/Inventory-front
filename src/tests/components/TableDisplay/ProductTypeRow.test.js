import React from 'react';
import '../../testHelpers/context';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import * as device from 'react-device-detect';

import ProductTypeRow from '../../../Components/TableDisplay/Row/ProductTypeRow/ProductTypeRow';
import Title from '../../../Components/Title/Title';

const baseProps = {
    config: {main : ['name', 'location', 'qty']},
    data: {
        name: 'Blue dogs',
        location: 'Dundee',
        qty: '12345678',
        price: '187.56',
        code: 'abc123',
        url: 'test.jpg'
    }
}

describe('<ProductTypeRow/> test suite', ()=>{
    it('matches snapshot', ()=>{
        const comp = renderer.create(<ProductTypeRow {...baseProps}/>);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });
    
    it('exists', ()=>{
        const wrapper = shallow(<ProductTypeRow {...baseProps} />);
        expect(wrapper.exists).toBeTruthy();
    });

    describe('isMobileOnly === false tests', ()=>{
        const orig = device.isMobileOnly;
        let wrapper;

        beforeEach(()=>{
            device.isMobileOnly = false;
            wrapper = mount(<ProductTypeRow {...baseProps}/>);
        });

        afterAll(()=>{
            device.isMobileOnly = orig;
        });
        
        it('renders 3 <div>s with classNames: "productTypeRow", "productTypeRow__title", "productTypeRow__cellCont"', ()=>{
            const tempW = shallow(<ProductTypeRow {...baseProps}/>);
            const div = tempW.find('div');
            expect(div.length).toBe(3);
            expect(div.at(0).props()).toHaveProperty('className', 'productTypeRow');
            expect(div.at(1).props()).toHaveProperty('className', 'productTypeRow__title');
            expect(div.at(2).props()).toHaveProperty('className', 'productTypeRow__cellCont');
        });
        
        it('renders a <Title/> component', ()=>{
            const t = wrapper.find(Title);
            expect(t.length).toBe(1);
            expect(t.props()).toHaveProperty('message', baseProps.data[baseProps.config.main[0]]);
            expect(t.props()).toHaveProperty('size', 'h4');
        });
        
        it('renders a <p> element with text', ()=>{
            expect(wrapper.find('p').length).toBe(1);
            expect(wrapper.find('p').text()).toEqual('location:Dundee');
        });
    });

    describe('isMobileOnly === true tests', ()=>{
        const orig = device.isMobileOnly;
        let wrapper;

        beforeEach(()=>{
            device.isMobileOnly = true;
            wrapper = mount(<ProductTypeRow {...baseProps}/>);
        });

        afterAll(()=>{
            device.isMobileOnly = orig;
        });

        it('only renders 2 <div> elements with classNames: productTypeRow and productTypeRow__title', ()=>{
            const tW = shallow(<ProductTypeRow {...baseProps}/>);
            const div = tW.find('div');
            expect(div.length).toBe(2);
            expect(div.at(0).props()).toHaveProperty('className', 'productTypeRow');
            expect(div.at(1).props()).toHaveProperty('className', 'productTypeRow__title');
        });
        
        it('renders a <Title/> component', ()=>{
            expect(wrapper.find(Title).length).toBe(1);
        });

        it('no <p> elements are rendered', ()=>{
            expect(wrapper.find('p').length).toBe(0);
        });
    });
});