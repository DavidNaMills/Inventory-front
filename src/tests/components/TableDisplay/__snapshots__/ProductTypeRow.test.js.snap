// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTypeRow/> test suite matches snapshot 1`] = `
<div
  className="productTypeRow"
>
  <div
    className="productTypeRow__title"
  >
    <div>
      <h4
        className="title_h4 "
      >
        Blue dogs
      </h4>
    </div>
  </div>
  <div
    className="productTypeRow__cellCont"
  >
    <p>
      <span>
        Locations
        :
      </span>
      Dundee
    </p>
  </div>
</div>
`;
